Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/NonCompositedFrameRenderer.cpp

```diff
@@ -71,6 +74,28 @@ NonCompositedFrameRenderer::~NonCompositedFrameRenderer()
     m_surface->willDestroyCompositingRunLoop();
 }
 
+void NonCompositedFrameRenderer::addDirtyRect(const IntRect& rect)
+{
+#if ENABLE(DAMAGE_TRACKING)
+    if (m_frameDamage)
+        m_frameDamage->add(rect);
+#else
+    UNUSED_PARAM(rect);
+#endif
+}
+
+#if ENABLE(DAMAGE_TRACKING)
+void NonCompositedFrameRenderer::resetFrameDamage()
+{
+    Ref webPage = m_webPage.get();
+    if (webPage->corePage()->settings().propagateDamagingInformation()) {
```

Comment:
```suggestion
    if (Ref webPage = m_webPage.get(); webPage->corePage()->settings().propagateDamagingInformation()) {
```

There is a chance that the promotion from WeakPtr to Ref fails, so it should be checked (or do `if (webPage && ...)` if you feel that's cleaner as the line is already long)